build/scripts: add script to build `bazel` on s390x

rickystewart opened the pull request:
Epic: CRDB-21133

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146119)
<!-- Reviewable:end -->

jlinder commented on licenses/APACHE-v2.0.txt:
This file should be named specifically to reference the project the code comes from, even though this is a generic license. See [these instructions](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/4246536270/How+to+copy+open-source+code+into+the+cockroach+repository#Naming-the-file-in-the-cockroach-licenses%2F-directory).

jlinder commented on build/scripts/build-bazel-s390x.sh:
```diff
@@ -0,0 +1,276 @@
+#
+# Use of this source code is governed by a LICENSE_NAME-style
+# license that can be found in licenses/APACHE-v2.0.txt.
+# LICENSE: Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
```
This line isn't necessary because the license is placed in the file in the `licenses/` directory.

jlinder commented on build/scripts/build-bazel-s390x.sh:
```diff
@@ -0,0 +1,276 @@
+#!/bin/bash
+# © Copyright IBM Corporation 2024
+#
+# Use of this source code is governed by a LICENSE_NAME-style
+# license that can be found in licenses/APACHE-v2.0.txt.
```
```suggestion
# Use of this source code is governed by the Apache License, Version 2.0,
# which can be found in licenses/APACHE-v2.0.txt.
```

jlinder commented on build/scripts/build-bazel-s390x.sh:
```diff
@@ -0,0 +1,276 @@
+# included in the /LICENSE file.
+#
+# This code originated in https://github.com/linux-on-ibm-z/scripts/blob/master/CockroachDB/24.1.6/build_crdb.sh, SHA d2d885b3b9edae9d0e69383c18475485149875b1
+#
```
I'm suggesting leaving a blank line between the license header and the rest of the file to clearly delineate the license header comment from the first non-license comment in the file. I do see this is a change from the original file.

```suggestion

```

jlinder reviewed (changes requested):
With these changes, this PR will follow our current license attribution practices.

rickystewart committed:
build/scripts: copy IBM script to build bazel 7.4.1

Copy from https://raw.githubusercontent.com/linux-on-ibm-z/scripts/refs/heads/master/Bazel/7.4.1/build_bazel.sh

Epic: CRDB-21133
Release note: None

rickystewart committed:
build/scripts: add script to build `bazel` on s390x

Bring the script up-to-date to build our current version of `bazel`
using our fork instead of patches.

Epic: CRDB-21133

Release note: None

jlinder reviewed (approved):
LGTM
